### PR TITLE
First Version of GameManager

### PR DESCRIPTION
**Pull request by EranDaniel98**

### CODE REVIEW

File: GameManager/GameManager.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+namespace GameManager
+{
+    class GameManager
+    {
+        public Player[] players;
+        private Board board;
+        private Server server;
+        private Dictionary<string, Player> EndGameOptions;
+
+        public GameManager(Server server, Board board, Player[] players)
```
Comment (NitsanLitov) [inline GameManager/GameManager.cs:18]: game manager should create a board

File: GameManager/GameManager.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            this.server = server;
+
+            this.EndGameOptions = new Dictionary<string, Player>();
+        }
+
+        void StartGame()
+        {
+            // server - update all that a piece moved
+            // server - Update the movement of specific player
+            bool is_running = true;
```
Comment (NitsanLitov) [inline GameManager/GameManager.cs:31]: give a better name

File: GameManager/GameManager.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            while (is_running)
+            {
+                foreach (Player p in players)
+                {
+                    Dictionary<(char, int), List<(char, int)>> movements = board.GetColorMovementOptions(p.Color); //piece location -> all the movement options
+                    if (movements.Keys.Count == 0)
+                    {
+                        if (board.IsKingThreatened(p)) // checkmate
```
Comment (NitsanLitov) [inline GameManager/GameManager.cs:39]: IsKingThreatened should get color

File: GameManager/GameManager.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                            this.EndGameOptions["checkmate"] = p;
```
Comment (NitsanLitov) [inline GameManager/GameManager.cs:40]: what if its mate? why is the dictionary empty?

Comment (NitsanLitov) [inline GameManager/GameManager.cs:40]: all messages should be const

Comment (NitsanLitov) [inline GameManager/GameManager.cs:40]: add comment to code:
// ToDo: when there is more then 2 players state which player won

File: GameManager/GameManager.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        public GameManager(Server server, Board board, Player[] players)
+        {
+            this.players = players;
+            this.board = board;
+            this.server = server;
+
+            this.EndGameOptions = new Dictionary<string, Player>();
+        }
+
+        void StartGame()
```
Comment (NitsanLitov) [inline GameManager/GameManager.cs:27]: always use this.

File: GameManager/GameManager.cs (hunk 3 above)
Comment (EranDaniel98) [reply GameManager/GameManager.cs:39]: Yep, you are right!

File: GameManager/GameManager.cs (hunk 4 above)
Comment (EranDaniel98) [reply GameManager/GameManager.cs:40]: 🤘🏻

**Pull request merged**